cylc/uiserver/resolvers.py

```diff
@@ -215,7 +217,7 @@ async def nodes_mutator(self, info, *m_args):
             flow[WORKFLOW].id
             for flow in await self.get_workflows_data(w_args)]
         if not w_ids:
-            return 'Error: No matching Workflow'
+            return [{'error': 'No matching workflows'}]
```

Comment:
I think this should be "error**s**" with the message in a map:

```suggestion
            return [{
                'errors': [{'message': 'No matching workflows'}]
            }]
```

http://spec.graphql.org/October2021/#sec-Errors

(If that doesn't play nicely with the UI [most likely my fault] we can fix later).

Reply:
I think this will require some follow-up work.

At the moment this response is [going through the mutator function twice](https://matrix.to/#/!SDKYnTgYyOXNBpJySp:matrix.org/$1636136658135167ecqpg:matrix.org?via=matrix.org), so even if you apply that suggestion we get
```json
{
    "data": {
        "<mutation>": {
            "result": [
                {
                    "errors": [{"message": "No matching workflows"}]
                }
            ]
        }
    }
}
```
Whereas what we should be getting is just the `errors`, no `data`:
```json
{
    "errors": [{"message": "No matching workflows"}]
}
```
Another possibility is we treat the no-match result as a non-error, e.g.  
```json
{
    "data": {
        "<mutation>": {
            "result": [
                {
                    "id": "whatever",
                    "response": [false, "No matching workflows"]
                }
            ]
        }
    }
}
```
or something like that

Reply:
Yeah, we could raise a GraphQL error.. But I'm not sure this warrants it, it's just information about the result of a command (it's not like the command itself needs to fail, although you could argue it).

You wouldn't get an ID back (unless you use your args, but that could be more than one)..

Reply:
Ah, ok, this is the generic response and not a GraphQL error.

In which case shouldn't it be a string rather than a map?